include/aspect/boundary_composition/ascii_data.h

```diff
@@ -0,0 +1,98 @@
+/*
+  Copyright (C) 2011 - 2014 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__boundary_composition_ascii_data_h
+#define __aspect__boundary_composition_ascii_data_h
+
+#include <aspect/boundary_composition/interface.h>
+
+#include <aspect/simulator_access.h>
+#include <aspect/utilities.h>
+
+
+namespace aspect
+{
+  namespace BoundaryComposition
+  {
+    using namespace dealii;
+
+    /**
+     * A class that implements prescribed boundary conditions
+     * determined from a AsciiData input file.
+     *
+     * @ingroup BoundaryCompositions
+     */
+    template <int dim>
+    class AsciiData : public Utilities::AsciiDataBoundary<dim>, public Interface<dim>
+    {
+      public:
+        /**
+         * Empty Constructor.
+         */
+        AsciiData ();
+
+        /**
+         * Initialization function. This function is called once at the
+         * beginning of the program. Checks preconditions.
+         */
+        virtual
+        void
+        initialize ();
```

Comment:
I don't think this function needs to be virtual -- you're not defining an interface here that you expect the derived classes to override.